domains/pallets/messenger/src/lib.rs

```diff
@@ -379,15 +395,45 @@ mod pallet {
                 Call::relay_message { msg: xdm } => {
                     let ValidatedRelayMessage {
                         msg,
+                        next_nonce,
                         should_init_channel: _,
                     } = Self::do_validate_relay_message(xdm)?;
-                    let provides_tag = (msg.dst_domain_id, msg.channel_id, msg.nonce);
-                    unsigned_validity("MessengerInbox", provides_tag)
+
+                    let mut valid_tx_builder = ValidTransaction::with_tag_prefix("MessengerInbox");
+                    // Only add the requires tag if the msg nonce is in future
+                    if msg.nonce > next_nonce {
+                        valid_tx_builder = valid_tx_builder.and_requires((
+                            msg.dst_domain_id,
+                            msg.channel_id,
+                            msg.nonce - Nonce::one(),
+                        ));
+                    };
+                    valid_tx_builder
+                        .priority(TransactionPriority::MAX)
+                        .longevity(TransactionLongevity::MAX)
+                        .and_provides((msg.dst_domain_id, msg.channel_id, msg.nonce))
+                        .propagate(true)
+                        .build()
                 }
                 Call::relay_message_response { msg: xdm } => {
-                    let msg = Self::do_validate_relay_message_response(xdm)?;
-                    let provides_tag = (msg.dst_domain_id, msg.channel_id, msg.nonce);
-                    unsigned_validity("MessengerOutboxResponse", provides_tag)
+                    let (msg, next_nonce) = Self::do_validate_relay_message_response(xdm)?;
+
+                    let mut valid_tx_builder =
+                        ValidTransaction::with_tag_prefix("MessengerOutboxResponse");
+                    // Only add the requires tag if the msg nonce is in future
+                    if msg.nonce > next_nonce {
```

Comment:
It is used to check if the msg is the immediate next message according to the current channel nonce state.

For example, suppose the current channel nonce is 1, the `msg1` won't have `requires` tag as its nonce is equal to the channel nonce thus it is the immediate next message and it will also have a `provides` tag `nonce1`. The `msg2`will have  `requires` tag `nonce1` and `provides` tag `nonce2`, since its `requires` match `msg1`'s `provides` it will be added to the tx pool ready queue after `msg1`.